### ISSUE TITLE
Adding label to a pod causes error

### ISSUE THREAD

**Issue by gaktive** [rancher/dashboard#8074]
Internal reference: SURE-5824
Reported in 2.7.0

Issue description:
Adding label to a pod via Rancher GUI causes error.

Repro steps:

step 1
Create a pod by kubectl: `$ kubectl run nginx-pod --image=nginx`
 
step 2
- Add a Label to the pod via Rancher GUI:
 - Cluster Manager -> local cluster -> Workload -> pods -> Edit Config -> Pod tab -> Labels & Annotations
- Push Add Label button: `key=foo, value=bar`
- Save
 
error occurs (error.png)
![error.png](https://zube.io/files/rancher/f6224c0ea57d0963bfb3944fcc2a2138-error.png)

```
Pod "nginx-pod" is invalid: spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)
core.PodSpec{
... // 11 identical fields
NodeName: "35.74.247.85",
SecurityContext: &{},

ImagePullSecrets: nil,
+ ImagePullSecrets: []core.LocalObjectReference{},
Hostname: "",
Subdomain: "",
SetHostnameAsFQDN: nil,
Affinity: nil,
+ Affinity: &core.Affinity{},
SchedulerName: "default-scheduler",
Tolerations: { {Key: "node.kubernetes.io/not-ready", Operator: "Exists", Effect: "NoExecute", TolerationSeconds: &300}
, {Key: "node.kubernetes.io/unreachable", Operator: "Exists", Effect: "NoExecute", TolerationSeconds: &300}},
... // 11 identical fields
}
```

**Comment by gaktive** [comment]
@scures saw this when trying to update environment variables in Epinio:
`Pod "epinio-server-5d5845745d-49ccj" is invalid: spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)`

This may be fixed in 2.7.2 elsewhere via Epinio work but this could be tied to a Helm deployment done not in the UI originally.

**Comment by Josh-Diamond** [comment]
Ticket #8074 - Test Results - ✅ 
---

With Docker on a single-node instance:

Reproduced w/ Rancher `v2.7.0`:
1. Fresh install of rancher `v2.7.0`
2. Navigate to `Cluster Explorer > KubeShell` of local cluster
3. Using kubectl imperative command, create a pod:
```kubectl
kubectl run <pod_name> --image=nginx
```
4. Once pod is created, navigate to `Cluster Explorer > Workloads > Pods`, find the pod and `edit config`
5. Via RancherUI, add a new pod label and then `save`
6. Reproduced - [`see screenshot`]

<img width="1638" alt="Screenshot 2023-03-14 at 11 04 34 AM" src="https://user-images.githubusercontent.com/46494969/225098079-49f788ad-3a64-43f1-b8fc-36cd8741d0e8.png">

---

Verified w/ Rancher `v2.7-211409c9a689d8ee4d801ed1bce1d887b0fb4b00-head`:
1. Fresh install of rancher `v2.7-head`
2. Navigate to `Cluster Explorer > KubeShell` of local cluster
3. Using kubectl imperative command, create a pod:
```kubectl
kubectl run <pod_name> --image=nginx
```
4. Once pod is created, navigate to `Cluster Explorer > Workloads > Pods`, find the pod and `edit config`
5. Via RancherUI, add a new pod label and then `save`
6. Verified - pod label successfully added via Rancher UI
